source/connectors/kafka-source.txt

```diff
@@ -0,0 +1,265 @@
+.. _kafka-source:
+
+============================
+Kafka Source Connector Guide
+============================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+The MongoDB Kafka Source Connector moves data from a MongoDB replica set
+into a Kafka cluster. The connector configures and consumes :manual:`change
+stream event documents </changeStreams>` and publishes them to a topic.
+
+Change streams, a feature introduced in MongoDB 3.6, generate event
+documents that contain changes to data stored in MongoDB in real-time and
+provide guarantees of durability, security, and idempotency. You can configure
+change streams to observe changes at the **collection**, **database**, or
+**deployment** level. See `An Introduction to Change Streams
+<https://www.mongodb.com/blog/post/an-introduction-to-change-streams>`_
+for more information.
+
+.. note::
+
+   Change streams require a replicaSet or a sharded cluster using replicaSets.
+
+Change Stream Event Document Format
+-----------------------------------
+
+A change stream event document contains several fields that describe the
+change:
+
+* The top-level ``_id`` field is used as the :manual:`resume token
+  </changeStreams/#change-stream-resume>` which is
+  used to start a change stream from a specific point in time.
+
+* The ``operationType`` field identifies the type of change represented in
+  the change stream document. Possible values include: "insert", "update",
+  "replace", "delete", "invalidate", "drop", "dropDatabase", and "rename".
+
+* The ``fullDocument`` field contents depend on the operation as follows:
+
+  - For insert and replace operations, it contains the new document being
+    inserted or replacing the existing document.
+  - For update operations, it contains the complete document that is being
```

Comment:
Interesting choice of wording. Where did you get this from? Not a nit.

Reply:
If you're referring to "some point in time", it is from this:
https://github.com/mongodb/mongo-kafka/blob/master/docs/source.md
![Screen Shot 2019-11-01 at 3 00 13 PM](https://user-images.githubusercontent.com/52428683/68049058-54e53400-fcb8-11e9-960d-e06fc4a86ad8.png)

Reply:
If there's a better way to phrase this or if you think we need to find an external source to provide additional detail, let me know.

Reply:
Nope, those is the best way to phrase it.